### PR TITLE
Support all gh-got options instead of only the token

### PR DESCRIPTION
**Pull request by millette**
Replaces #9 (sorry for my lack of git-fu).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+const defaultOptions = {
+	headers: {
+		'user-agent': 'https://github.com/sindresorhus/gh-user'
+	}
+};
+
+module.exports = async (username, options) => {
 	if (typeof username !== 'string' || !username) {
 		throw new Error('`username` required');
 	}
```
Comment (sindresorhus) [inline index.js]: The options check can be simplified by doing an early type check instead:

```js
if (options !== 'object') {
	throw new TypeError('`options` should be an object');
}
```

File: index.js [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
-		headers: {
-			'user-agent': 'https://github.com/sindresorhus/gh-user'
+	if (typeof options === 'object') {
+		options = {...defaultOptions, ...options};
+	} else {
+		if (options !== undefined) {
+			throw new TypeError('`options` should be an object of gh-got options');
 		}
-	});
+		options = {...defaultOptions};
```
Comment (sindresorhus) [inline index.js]: No point in using spreading here. You can just do a normal assignment.

File: index.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
 	if (typeof username !== 'string' || !username) {
 		throw new Error('`username` required');
 	}
 
-	const result = await ghGot(`users/${username}`, {
-		token,
-		headers: {
-			'user-agent': 'https://github.com/sindresorhus/gh-user'
+	if (typeof options === 'object') {
+		options = {...defaultOptions, ...options};
```
Comment (sindresorhus) [inline index.js]: Spreading only work on one level, so the `header` option, for example, will not be correctly handled. You should use https://github.com/sindresorhus/got/blob/master/readme.md#gotmergeoptionsparentoptions-newoptions instead.

File: index.js (hunk 2 above)
Comment (millette) [reply index.js]: I changed the signature to use a default {} options. It simplified everything.

**Pull request merged**